source/materials/XenonGasProperties.cc

```diff
@@ -212,12 +212,23 @@ G4double XenonGasProperties::ELLightYield(G4double field_strength) const
 
   // Y/x = (a E/p - b) p,
   // where Y/x is the number of photons per unit lenght (cm),
-  // E is the electric field strength, p is the pressure, and a and b
-  // are empirically determined constants:
+  // E is the electric field strength, p is the pressure
+  // a and b are empirically determined constants depending on pressure.
+  // Values for different pressures are found in: Freitas-2010
+  // Physics Letters B 684 (2010) 205–210
 
-  const G4double a = 140. / kilovolt;
   const G4double b = 116. / (bar*cm);
+  G4double a = 140. / kilovolt;
 
+  // Updating the slope
+  if (pressure_ >= 2. * bar) a = 141. / kilovolt;
+  if (pressure_ >= 3. * bar) a = 141. / kilovolt;
```

Comment:
For 2 and 3 bar the constant is the same. Why don't you drop the line with 3 bar?

Reply:
I was trying to replicate all the values that appeared in Freitas paper (the one we are following), and the values for both pressures appeared in the list. I thought it was a good idea to maintain all the values although they could be the same.

But I don't have a strong opinion on this, so we can delete it if you think it's better.

Reply:
It's a superfluous line of code, I think it's better to drop it.

Reply:
Done

Reply:
There's something to be said for directly reflecting in the code, the structure of a paper that the code follows. But, in that case, it's vital that you add comments explaining why the code is doing something that appears odd, lest readers waste valuable time and effort on wondering why you are doing something apparently pointless.

Reply:
I fully agree. I thought my comments at the beginning of the function ware enough.

Reply:
Ah, sorry, those comments were not visible in the little extract that GitHub showed me.